* Sequencing

- https://www.genome.gov/sequencingcostsdata/

* Sequencing read data bases

- [[https://www.ncbi.nlm.nih.gov/geo/][NCBI GEO]]
- [[https://www.ncbi.nlm.nih.gov/sra][NCBI Short Read Archive]]
- [[https://www.ebi.ac.uk/arrayexpress/][EBI ArrayExpress]]

* (Short) read mappers

There is plethora of different short read mappers. A good and well
maintained overview can be found [[https://www.ebi.ac.uk/~nf/hts_mappers/][here]].

Mapper we use:
- [[https://github.com/alexdobin/STAR][STAR]]
- [[http://www.bioinf.uni-leipzig.de/Software/segemehl/][segemehl]]
- [[http://bio-bwa.sourceforge.net/][BWA]]

An alternative to actually map reads and quantify them is [[https://pachterlab.github.io/kallisto/][kallisto]]
which performs a pseudoalignment based on k-mers.

* File Formats

http://www.ensembl.org/info/website/upload/bed.html

- BED http://www.ensembl.org/info/website/upload/bed.html
- GFF2 aka GTF
- GFF3 
- FASTQ
- SAM/BAM/CRAM

* SRA tools
* cutadapt
* FastQC
* samtools/bamtools
* bedtools
* picard
* DGEA 
** DESeqq
** EdgeR
** voom
** RUVSeq
* GSEA
* Genom browsers

