* Sequencing

- https://www.illumina.com/Documents/products/Illumina_Sequencing_Introduction.pdf

- https://www.genome.gov/sequencingcostsdata/

- Good review about sequencing technologies and protocols: Reuter JA,
  Spacek DV, Snyder MP. High-throughput sequencing technologies. Mol
  Cell. 2015 May 21;58(4):586-97. doi: [[https://doi.org/10.1016/j.molcel.2015.05.004][10.1016/j.molcel.2015.05.004]].


- http://www.nature.com/nbt/journal/v30/n11/full/nbt.2421.html

* Sequencing read data bases

- [[https://www.ncbi.nlm.nih.gov/geo/][NCBI GEO]]
- [[https://www.ncbi.nlm.nih.gov/sra][NCBI Short Read Archive]]
- [[https://www.ebi.ac.uk/arrayexpress/][EBI ArrayExpress]]

* (Short) read mappers

There is plethora of different short read mappers. A good and well
maintained overview can be found [[https://www.ebi.ac.uk/~nf/hts_mappers/][here]].

Mapper we use:
- [[https://github.com/alexdobin/STAR][STAR]]
- [[http://www.bioinf.uni-leipzig.de/Software/segemehl/][segemehl]]
- [[http://bio-bwa.sourceforge.net/][BWA]]

An alternative to actually map reads and quantify them is [[https://pachterlab.github.io/kallisto/][kallisto]]
which performs a pseudoalignment based on k-mers.

* File Formats

There are several file format that are commonly used for the analysis
of high-throughput sequencing data

http://www.ensembl.org/info/website/upload/bed.html

- SAM and it BAM ([[https://samtools.github.io/hts-specs/SAMv1.pdf][specification as PDF]])
- [[https://www.ebi.ac.uk/ena/software/cram-toolkit][CRAM]] ([[https://samtools.github.io/hts-specs/CRAMv3.pdf][specification as PDF]])
- [[http://www.ensembl.org/info/website/upload/bed.html][BED ]]
- [[http://www.ensembl.org/info/website/upload/gff.html][GFF2]] (aka GTF - Gene transfer format)
- [[http://gmod.org/wiki/GFF3][GFF3]]
- [[https://en.wikipedia.org/wiki/FASTA_format][FASTA]]
- [[https://en.wikipedia.org/wiki/FASTQ_format][FASTQ]]
- [[http://www.ensembl.org/info/website/upload/wig.html][Wiggle]] / [[https://genome.ucsc.edu/goldenpath/help/bigWig.html][BigWig]]
- [[https://genome.ucsc.edu/goldenpath/help/bedgraph.html][BedGraph]]
- [[http://www.1000genomes.org/wiki/Analysis/vcf4.0/][VCF]] (Variant Call Format)

* SRA tools

- [[https://github.com/ncbi/sra-tools][SRA tools]]

* cutadapt

http://cutadapt.readthedocs.io/en/stable/index.html

* FastQC

http://www.bioinformatics.babraham.ac.uk/projects/fastqc/

* SAMtools / BAMtools

- [[http://www.htslib.org/][SAMtools]]
- [[https://github.com/pezmaster31/bamtools][BAMtools]]
- [[https://www.ebi.ac.uk/ena/software/cram-toolkit][CRAMtools]]

* bedtools

[[http://bedtools.readthedocs.io][- bedtools]]

* Cheat Sheet For One-Based Vs Zero-Based Coordinate Systems

- [[https://www.biostars.org/p/84686/][Cheat Sheet For One-Based Vs Zero-Based Coordinate Systems]]

* Picard

https://broadinstitute.github.io/picard/

* Differential Gene expression analysis

- Good overview article: Schurch NJ, Schofield P, Gierli≈Ñski M, Cole
  C, Sherstnev A, Singh V, Wrobel N, Gharbi K, Simpson GG, Owen-Hughes
  T, Blaxter M, Barton GJ. How many biological replicates are needed
  in an RNA-seq experiment and which differential expression tool
  should you use? RNA. 2016 Jun;22(6):839-51. doi:
  [[http://dx.doi.org/10.1261/rna.053959.115][10.1261/rna.053959.115]].
- Sequencing depth vs. Replicates: 
- [[https://bioconductor.org/packages/release/bioc/html/DESeq2.html][DESeq2]]
- [[https://bioconductor.org/packages/release/bioc/html/edgeR.html][EdgeR]]
- [[https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29][voom]]
- [[https://bioconductor.org/packages/release/bioc/html/RUVSeq.html][RUVSeq]] (remove unwanted variation) to use before other tools for DGEA
* Gene set enrichment analysis tools

- [[https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html][clusterProfiler]]
- [[https://bioconductor.org/packages/release/bioc/html/topGO.html][topGO]]
- [[http://software.broadinstitute.org/gsea/index.jsp][GSEA]]

* Transcriptional start site prediction based on dRNA-Seq

- [[http://it.inf.uni-tuebingen.de/?page_id=190][TSSpredator]]
- [[http://nibiru.tbi.univie.ac.at/TSSAR/][TSSAR]]
- [[http://www.clipz.unibas.ch/downloads/TSSer/index.php][TSSer]]

* Genom browsers

- [[http://bioviz.org/igb/index.html][Integrated genome browser]] (IGB)
- [[http://software.broadinstitute.org/software/igv/][Integrative Genomics Viewer]] (IGV)
- [[http://genomeview.org/][GenomeView]]
- [[http://jbrowse.org/][JBrowse]]
- [[http://www.biodalliance.org/][Biodalliance]]

* HiC data analysis

- [[https://nservant.github.io/HiC-Pro/][HiC-Pro]]
- [[https://bioconductor.org/packages/release/bioc/html/HiTC.html][HiTC]]
- [[http://aidenlab.org/juicer/docs.html][Juicer]] / [[http://www.aidenlab.org/juicebox/][JuiceBox]]
* Misc

- [[https://statquest.org/2015/07/09/rpkm-fpkm-and-tpm-clearly-explained/][RPKM, FPKM and TPM, clearly explained]]
