@article{karlebach_modelling_2008,
	title = {Modelling and analysis of gene regulatory networks},
	volume = {9},
	issn = {1471-0072},
	url = {https://www.nature.com/nrm/journal/v9/n10/full/nrm2503.html#top},
	doi = {10.1038/nrm2503},
	abstract = {Gene regulatory networks have an important role in every process of life, including cell differentiation, metabolism, the cell cycle and signal transduction. By understanding the dynamics of these networks we can shed light on the mechanisms of diseases that occur when these cellular processes are dysregulated. Accurate prediction of the behaviour of regulatory networks will also speed up biotechnological projects, as such predictions are quicker and cheaper than lab experiments. Computational methods, both for supporting the development of network models and for the analysis of their functionality, have already proved to be a valuable research tool.},
	language = {en},
	number = {10},
	urldate = {2017-06-25},
	journal = {Nature Reviews Molecular Cell Biology},
	author = {Karlebach, Guy and Shamir, Ron},
	month = oct,
	year = {2008},
	pages = {770--780}
}

@article{weissgerber_beyond_2015,
	title = {Beyond {Bar} and {Line} {Graphs}: {Time} for a {New} {Data} {Presentation} {Paradigm}},
	volume = {13},
	shorttitle = {Beyond {Bar} and {Line} {Graphs}},
	url = {http://dx.doi.org/10.1371/journal.pbio.1002128},
	doi = {10.1371/journal.pbio.1002128},
	abstract = {A systematic review of research articles reveals widespread poor practice in the presentation of continuous data. The authors recommend training for investigators and supply templates for easy use.},
	number = {4},
	urldate = {2015-05-12},
	journal = {PLoS Biol},
	author = {Weissgerber, Tracey L. and Milic, Natasa M. and Winham, Stacey J. and Garovic, Vesna D.},
	month = apr,
	year = {2015},
	keywords = {visualization},
	pages = {e1002128},
}

@article{nitzan_integration_2017,
	title = {Integration of {Bacterial} {Small} {RNAs} in {Regulatory} {Networks}},
	volume = {46},
	url = {https://doi.org/10.1146/annurev-biophys-070816-034058},
	doi = {10.1146/annurev-biophys-070816-034058},
	abstract = {Small RNAs (sRNAs) are central regulators of gene expression in bacteria, controlling target genes posttranscriptionally by base pairing with their mRNAs. sRNAs are involved in many cellular processes and have unique regulatory characteristics. In this review, we discuss the properties of regulation by sRNAs and how it differs from and combines with transcriptional regulation. We describe the global characteristics of the sRNAâ€“target networks in bacteria using graph-theoretic approaches and review the local integration of sRNAs in mixed regulatory circuits, including feed-forward loops and their combinations, feedback loops, and circuits made of an sRNA and another regulator, both derived from the same transcript. Finally, we discuss the competition effects in posttranscriptional regulatory networks that may arise over shared targets, shared regulators, and shared resources and how they may lead to signal propagation across the network.},
	number = {1},
	urldate = {2017-07-01},
	journal = {Annual Review of Biophysics},
	author = {Nitzan, Mor and Rehani, Rotem and Margalit, Hanah},
	year = {2017},
	pmid = {28532217},
	pages = {131--148},
}
